repository_service_tuf/cli/admin/metadata.py

```diff
@@ -0,0 +1,498 @@
+        if not current_root.remove_key(name):
+            console.print(
+                "\n", f":cross_mark: [red]Failed[/]: key {name} is not in root"
+            )
+        else:
+            console.print(f"Key with name/tag [yellow]{name}[/] removed\n")
+            if len(current_root.root_keys) < 1:
+                console.print("No keys are left for removal.")
+                break
+
+
+def _keys_additions(current_root: RootInfo):
+    root_threshold = current_root.threshold
+    console.print(f"You need to have at least [cyan]{root_threshold}[/] keys.")
+    while True:
+        keys_table = _create_keys_table(
+            current_root, is_online_key_table=False, is_minimal=False
+        )
+        console.print("\nHere are the current root keys:")
+        console.print(keys_table)
+        response = prompt.Confirm.ask("\nDo you want to add a new key?")
+        if not response:
+            keys_amount = len(current_root.root_keys)
+            if keys_amount < root_threshold:
+                remaining_keys = root_threshold - keys_amount
+                console.print(
+                    f"You need to add an additional {remaining_keys} keys"
+                )
+                abort = prompt.Confirm.ask(
+                    "Do you want to abort the root metadata update",
+                )
+                if abort:
+                    raise click.ClickException(
+                        "Not enough keys to fulfill the threshold requirement"
+                    )
+                else:
+                    continue
+            else:
+                break
+
+        root_key: RSTUFKey = _get_key(Root.type)
+        if root_key.error:
+            console.print(f":cross_mark: [red]Failed[/]: {root_key.error}")
+            continue
+
+        if root_key == current_root.online_key:
+            console.print(
+                ":cross_mark: [red]Failed[/]: This is the current online key. "
+                "Cannot be added"
+            )
+            continue
+
+        if current_root.is_keyid_used(root_key.key["keyid"]):
+            console.print(":cross_mark: [red]Failed[/]: Key is already used")
+            continue
+
+        root_key.name = prompt.Prompt.ask(
+            "[Optional] Give a [green]name/tag[/] to the key"
+        )
+
+        current_root.add_key(root_key)
+
+
+def _get_positive_int_input(msg: str, input_name: str, default: Any) -> int:
+    input: int = 0
+    while input < 1:
+        input = prompt.IntPrompt.ask(msg, default=default, show_default=True)
+        if input < 1:
+            console.print(f"{input_name} must be at least 1")
+            continue
+
+    return input
+
+
+def _modify_expiration_policy(current_root: RootInfo):
+    console.print(markdown.Markdown(EXPIRY_CHANGES_MSG), width=100)
+    expiry_table = table.Table()
+    expiry_table.add_column("Role", justify="center")
+    expiry_table.add_column("Expiration Policy", justify="center")
```

Comment:
I was evaluating it. Let's handle the expiration not in the Metadata Update feature.
There is no reason for us to implement it during the metadata rotation.

We can handle it using the REST API  POST in `/api/v1/config` and specific feature in the CLI.

In the Metadata Update, we coudl:

```mermaid
flowchart TD
    root[Do you want to change the root metadata?] --> root_answer{Y/N}
    root_answer-->|Yes| root_expire[Do you want update root expiration?]
    root_answer-->|No| online_keys[Online keys...]
    root_expire --> root_expire_answer{Y/N}
    root_expire_answer -->|Yes| root_new_expire_days[Exprire of days from now: 365]
    root_expire_answer -->|No| root_keys[Root Keys ... ]


```

Reply:
Ahaa, so updating only root expiration and we can create another end point to update the automatic bump expiry.
Let's have a discussion about what we want to document in that issue.

Reply:
Now, when we won't update the roles using the online key we can change the wording from `expiration policy` to `expiration bump` as it will show clearly that the expiration will be actually bumped.

Reply:
I closed the old:
- https://github.com/repository-service-tuf/repository-service-tuf-worker/issues/287